lib/schemaUtils.js

```diff
@@ -2356,7 +2356,9 @@ module.exports = {
       }
       pmBody = this.convertToPmBody(reqBody, REQUEST_TYPE.ROOT, components, options, schemaCache);
       item.request.body = pmBody.body;
-      item.request.addHeader(pmBody.contentHeader);
+      if (!options.keepImplicitHeaders || (reqParams.header && reqParams.header.length === 0)) {
```

Comment:
This scenario fails to add an auto-detected `content-type` header even if there is no `content-type` header mentioned under parameters.

We can check specifically for `content-type` here instead of `(reqParams.header && reqParams.header.length === 0)` with following check.

```
!_.find(reqParams.header, (h) => { return _.toLower(_.get(h, 'name')) === 'content-type'; });
```

Reply:
Good remark, I changed the IF as suggested.